scripts/local-deploy.sh

```diff
@@ -78,89 +171,70 @@ cp "${repo_basedir}"/certificates/*.crt secrets/
-while [ ! "$(curl -s -o /dev/null -I -w "%{http_code}" --max-time 10 -H "Accept: application/json" -H "Content-Type: application/json" -X GET https://"${MACHINE_FQDN}":5000/api/v1/users)" = 401 ]; do
-    echo "waiting for portus to run..."
-    sleep 5s
-done
-
-if [ ! -f .portus_token ]; then
-    echo
-    echo "configuring portus via its API ..."
-json_data=$(cat <<EOF
-{
-    "user": {
-        "username": "$SERVICES_USER",
-        "email": "admin@swiss",
-        "password": "$SERVICES_PASSWORD"
-    }
-}
-EOF
-)
-    portus_token=$(curl -H "Accept: application/json" -H "Content-Type: application/json" -X POST \
-        -d "${json_data}" https://"$MACHINE_FQDN":5000/api/v1/users/bootstrap | jq -r .plain_token)
-    echo "${portus_token}" >> .portus_token
-
-json_data=$(cat <<EOF
-{
-    "registry": {
-        "name": "$MACHINE_FQDN",
-        "hostname": "$MACHINE_FQDN:5000",
-        "use_ssl": true
-    }
-}
-EOF
-)
-    curl -H "Accept: application/json" -H "Content-Type: application/json" -H "Portus-Auth: $SERVICES_USER:${portus_token}"  -X POST \
-        -d "${json_data}" https://"$MACHINE_FQDN":5000/api/v1/registries
-fi
+$psed -i -e "s/REGISTRY_DOMAIN=.*/REGISTRY_DOMAIN=$REGISTRY_DOMAIN/" .env
+$psed -i -e "s/S3_ACCESS_KEY_ID=.*/S3_ACCESS_KEY_ID=$SERVICES_PASSWORD/" .env
+$psed -i -e "s/S3_SECRET_ACCESS_KEY=.*/S3_SECRET_ACCESS_KEY=$SERVICES_PASSWORD/" .env
+$psed -i -e "s/S3_BUCKET=.*/S3_BUCKET=${S3_BUCKET}/" .env
+$psed -i -e "s/S3_ENDPOINT=.*/S3_ENDPOINT=${S3_ENDPOINT}/" .env
+make up
 popd
 
 # -------------------------------- MONITORING -------------------------------
 echo
 echo -e "\e[1;33mstarting monitoring...\e[0m"
 # set MACHINE_FQDN
 pushd "${repo_basedir}"/services/monitoring
+$psed -i -e "s/MONITORING_DOMAIN=.*/MONITORING_DOMAIN=$MONITORING_DOMAIN/" .env
 $psed -i -e "s|GF_SERVER_ROOT_URL=.*|GF_SERVER_ROOT_URL=https://$MACHINE_FQDN/grafana|" grafana/config.monitoring
 $psed -i -e "s|GF_SECURITY_ADMIN_PASSWORD=.*|GF_SECURITY_ADMIN_PASSWORD=$SERVICES_PASSWORD|" grafana/config.monitoring
 $psed -i -e "s|basicAuthPassword:.*|basicAuthPassword: $SERVICES_PASSWORD|" grafana/provisioning/datasources/datasource.yml
-$psed -i -e "s|--web.external-url=.*|--web.external-url=https://$MACHINE_FQDN/prometheus/'|" docker-compose.yml
+
+# if  WSL, we comment - /etc/hostname:/etc/host_hostname
+if [ is_WSL ] 
+then 
+    if [ ! $(grep -qEi  "#- /etc/hostname:/etc/nodename # don't work with windows" &> /dev/null docker-compose.yml) ]
+    then
+        $psed -i -e "s~- /etc/hostname:/etc/nodename # don't work with windows~#- /etc/hostname:/etc/nodename # don't work with windows~" docker-compose.yml
+    fi
+else
+    if [ $(grep  "#- /etc/hostname:/etc/nodename # don't work with windows" &> /dev/null docker-compose.yml) ]  
+    then
+        $psed -i -e "s~#- /etc/hostname:/etc/nodename # don't work with windows~- /etc/hostname:/etc/nodename # don't work with windows~" docker-compose.yml
+    fi
+fi
+
```

Comment:
is that why you have the is_WSL? I would not bother about windows...

Reply:
I am not sure why you have an else case ?? actually this is going to be not necessary with WSL2...